### ISSUE TITLE
Can I load Torch trained model file by Pytorch

### ISSUE THREAD

**Issue by erogol** [hughperkins/pytorch#5]
I want to use torch trained deep learning model from python. Is it able to do that?

**Comment by hughperkins** [comment]
Yes ... you can create a lua class, that trains your model, or loads an existing model, and call this from Python.  You can pass in new examples from python to the model, to predict on.  I would first have a look at [simpleexample/pybit.py](https://github.com/hughperkins/pytorch/blob/master/simpleexample/pybit.py) and [simpleexample/luabit.lua](https://github.com/hughperkins/pytorch/blob/master/simpleexample/luabit.lua) for an example of python code that loads a lua class, creates an object from it, and calls methods on it.  That lua class could load your model, and pass new examples to your model for prediction.

**Comment by hughperkins** [comment]
Actually, I will write an example for this :-)

**Comment by erogol** [comment]
Thanks :)

**Comment by hughperkins** [comment]
Ok, I didnt add prediction, but covers:
- building a network
- passing in training data, in batches
- training the network
- getting training loss and accuracy
- also handles: cpu, cuda, opencl :-)

51cbc09

https://github.com/hughperkins/pytorch/tree/master/examples/luamodel

**Comment by hughperkins** [comment]
Updated so you can pass numpy tensors in directly now 66bb797

**Comment by hughperkins** [comment]
added prediction in 9c05864

Also added methods `tensorOne.eq(tensorTwo)`, and `tensor.sum()` to PyTorch.xxxTensor objects

You'll need to `pip install docopt` to use runner.py now

**Comment by hughperkins** [comment]
Updated in 0a9e04a to convert the output predictions tensor to numpy array, before calculating number correct

**Comment by hughperkins** [comment]
To what extent does this do approximately what you want?  (The other approach is to instantiate nn modules directly from the python; depends what you want really)

**Comment by hughperkins** [comment]
You can load directly now, without needing to create a lua class, see https://github.com/hughperkins/pytorch/issues/15
